butane_core/src/sqlval.rs

```diff
@@ -625,6 +639,28 @@ impl FieldType for DateTime<chrono::offset::Utc> {
 #[cfg(feature = "datetime")]
 impl PrimaryKeyType for DateTime<chrono::offset::Utc> {}
 
+#[cfg(feature = "datetime")]
+impl_basic_from_sql!(NaiveDate, Date, Date);
+#[cfg(feature = "datetime")]
+impl ToSql for NaiveDate {
+    fn to_sql(&self) -> SqlVal {
+        SqlVal::Date(*self)
+    }
+    fn to_sql_ref(&self) -> SqlValRef<'_> {
+        SqlValRef::Date(*self)
+    }
+    fn into_sql(self) -> SqlVal {
+        SqlVal::Date(self)
+    }
+}
+#[cfg(feature = "datetime")]
+impl FieldType for NaiveDate {
+    const SQLTYPE: SqlType = SqlType::Date;
+    type RefType = str;
+}
+#[cfg(feature = "datetime")]
+impl PrimaryKeyType for NaiveDate {}
```

Comment:
would be good to add a test in butane/tests/basic.rs for using a date as a primary key field.